openwisp_firmware_upgrader/api/views.py

```diff
@@ -0,0 +1,80 @@
+from rest_framework import generics
+from swapper import load_model
+
+from .serializers import (
+    BatchUpgradeOperationSerializer,
+    BuildSerializer,
+    CategorySerializer,
+    FirmwareImageSerializer,
+)
+
+BatchUpgradeOperation = load_model('firmware_upgrader', 'BatchUpgradeOperation')
+Build = load_model('firmware_upgrader', 'Build')
+Category = load_model('firmware_upgrader', 'Category')
+FirmwareImage = load_model('firmware_upgrader', 'FirmwareImage')
+
+
+class OrgAPIMixin:
+    def get_queryset(self):
+        queryset = self.model.objects.all()
+        if not self.request.user.is_superuser:
+            user_org = self.request.user.openwisp_users_organization.get()
+            organization_filter = {self.organization_field: user_org}
+            queryset = queryset.filter(**organization_filter)
+        org = self.request.query_params.get('org', None)
+        try:
+            if org:
+                organization_filter = {self.organization_field + '__name': org}
+                queryset = queryset.filter(**organization_filter)
+        except Exception:
+            # django.core.exceptions.ValidationError when uuid is not valid
+            queryset = []
+        return queryset
+
+
+class BuildAPIListView(OrgAPIMixin, generics.ListCreateAPIView):
```

Comment:
I'd remove `API` from the name of the views since we're already in `api.py` and it's implicit.